From f8ba240ba0059833c2ac56a8a669164c4e08cdb2 Mon Sep 17 00:00:00 2001
From: Keyon Jie <yang.jie@linux.intel.com>
Date: Mon, 28 May 2018 13:43:18 +0800
Subject: [PATCH 1660/2328] ASoC: Intel: bxt-tdf8532: FIX: don't use
 add_dai_link() for SOF

We set ignore_machine for SOF soc platform driver, which will trigger
overriding FEs in soc_check_tplg_fes(), but this overriding may be
overidden again by add_dai_link() in bxt_tdf8532, e.g. platform_name
will be modified to be "0000:00:0e.0", which is not exist in SOF.

Here add #ifdef to bypass add_dai_link() for using the machine driver
with SOF to fix the overridden again issue.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
---
 sound/soc/intel/boards/bxt_tdf8532.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/intel/boards/bxt_tdf8532.c b/sound/soc/intel/boards/bxt_tdf8532.c
index f5c2a06..ffd9cc5 100644
--- a/sound/soc/intel/boards/bxt_tdf8532.c
+++ b/sound/soc/intel/boards/bxt_tdf8532.c
@@ -378,6 +378,7 @@ static struct snd_soc_dai_link broxton_tdf8532_dais[] = {
 	},
 };
 
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL)
 static int bxt_add_dai_link(struct snd_soc_card *card,
 			    struct snd_soc_dai_link *link)
 {
@@ -385,6 +386,7 @@ static int bxt_add_dai_link(struct snd_soc_card *card,
 	link->nonatomic = 1;
 	return 0;
 }
+#endif
 
 /* broxton audio machine driver for TDF8532 */
 static struct snd_soc_card broxton_tdf8532 = {
@@ -398,7 +400,9 @@ static struct snd_soc_card broxton_tdf8532 = {
 	.dapm_routes = broxton_tdf8532_map,
 	.num_dapm_routes = ARRAY_SIZE(broxton_tdf8532_map),
 	.fully_routed = true,
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL)
 	.add_dai_link = bxt_add_dai_link,
+#endif
 };
 
 static int broxton_tdf8532_audio_probe(struct platform_device *pdev)
-- 
2.7.4

